sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -119,4 +121,32 @@ object IntervalUtils {
       case _: IllegalArgumentException => null
     }
   }
+
+  /**
+   * Gets interval duration
+   *
+   * @param cal - the interval to get duration
+   * @param targetUnit - time units of the result
+   * @param daysPerMonth - the number of days per one month
+   * @return duration in the specified time units
+   */
+  def getDuration(
+      cal: CalendarInterval,
+      targetUnit: TimeUnit,
+      daysPerMonth: Int = 31): Long = {
+    val monthsDuration = Math.multiplyExact(daysPerMonth * DateTimeUtils.MICROS_PER_DAY, cal.months)
+    val result = Math.addExact(cal.microseconds, monthsDuration)
+    targetUnit.convert(result, TimeUnit.MICROSECONDS)
+  }
+
+  /**
+   * Checks the interval is negative
+   *
+   * @param cal - the checked interval
+   * @param daysPerMonth - the number of days per one month
+   * @return true if duration of the given interval is less than 0 otherwise false
+   */
+  def isNegative(cal: CalendarInterval, daysPerMonth: Int = 31): Boolean = {
```

Comment:
ditto, we can call it `isNegativeWatermark`.

Reply:
IMHO, this is confusing method name. Here, we do not check any watermarks. I would understand `isNegativeDelay`

Reply:
I agree, this isn't specific to one usage. If it lives in a general class a general name is better